library/vulnerabilities/path-traversal/checkUrlPathForPathTraversal.ts

```diff
@@ -0,0 +1,58 @@
+import { getRawUrlPath } from "../../helpers/getRawUrlPath";
+import { normalizeLikeURLConstructor } from "./normalizeLikeURLConstructor";
+
+const forbiddenPattern = /(?:^|[\\/])\.\.(?:[\\/]|$)/;
+
+/**
+ * Check if the URL path contains a path traversal attack
+ * A legitimate URL path should not contain ".." or "/../" or "\..\" or "\..\"
+ * https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.4
+ */
+export function checkUrlPathForPathTraversal(url: string | undefined): {
+  found: boolean;
+  payload?: string;
+} {
+  if (!url || url.length < 3) {
+    // Performance optimization, url most include at least 3 characters (/..)
+    return {
+      found: false,
+    };
+  }
+
+  const rawPath = getRawUrlPath(url);
+  if (rawPath.length < 3) {
+    // Performance optimization, we don't need to check for path traversal if the path is less than 3 characters
+    // Can happen if the URL has a query string
+    return {
+      found: false,
+    };
+  }
+
+  if (forbiddenPattern.test(rawPath)) {
+    return {
+      found: true,
+      payload: rawPath,
+    };
+  }
+
+  if (!url.includes("%")) {
```

Comment:
We could also add a length check? Since we expect at least two chars behind `%`?

Reply:
Nevermind already checked above 👍